cpp/include/raft/distance/distance.cuh

```diff
@@ -109,20 +108,20 @@ template <raft::distance::DistanceType distanceType,
           typename AccType,
           typename OutType,
           typename Index_ = int>
-void distance(const InType* x,
+void distance(raft::resources const& handle,
```

Comment:
This might be a breaking change, though it might not? (e.g. [comment here](https://github.com/rapidsai/raft/pull/1226/files#r1099603821)). We probably want to build cuml w/ this change at least to check.

Reply:
testing out w/ cuml here https://github.com/rapidsai/cuml/pull/5230

Reply:
needs https://github.com/rapidsai/cuml/pull/5230/commits/6f0d7fce048fd716a0e025882f79db241eb07d35 in cuml looks like